src/js/components/AppListComponent.jsx

```diff
@@ -70,13 +70,19 @@ var AppListComponent = React.createClass({
     var sortKey = state.sortKey;
     var props = this.props;
 
-    return lazy(state.apps)
+    var appsSequence = lazy(state.apps);
+
+    if (props.filterText != null && props.filterText != "") {
+      appsSequence = appsSequence
+        .filter(function (app) {
+          return app.id.indexOf(props.filterText) !== -1;
```

Comment:
Why are they misleading?
This makes sense f.ex. and should be faster than the native approach:
```
Lazy([1, 2, 3]).map(reciprocal).indexOf(0.5)
```
indexOf is resovling the sequence and produces a real value, you can see this here:
```
returns 	number
```